src/lumigo_tracer/parsers/utils.py

```diff
@@ -147,7 +147,7 @@ def recursive_json_join(d1: dict, d2: dict):
     for key in set(itertools.chain(d1.keys(), d2.keys())):
         value = d1.get(key, d2.get(key))
         if isinstance(value, dict):
-            d[key] = recursive_json_join(d1.get(key, {}), d2.get(key, {}))
```

Comment:
trim the non-existing branch fast

Reply:
nope. 5 lines above we trim the tree if we get None. returning `{}` makes it to never be trimmed...